experimental/LieAlgebras/src/WeylGroup.jl

```diff
@@ -1,29 +1,72 @@
-function coxeter_matrix(W::WeylGroup)
-  return cartan_to_coxeter_matrix(cartan_matrix(root_system(W)))
-end
-
 @doc raw"""
+    FPGroup(W::WeylGroup) -> FPGroup
     fp_group(W::WeylGroup) -> FPGroup
 
 Construct a group of type `FPGroup` that is isomorphic to `W`.
 
-The `FPGroup` will be the quotient of a free group with the same rank as `W`,
-where we have the natural 1-to-1 correspondence of generators, modulo the Coxeter relations of `W`.
+If one needs the isomorphism then [`isomorphism(::Type{FPGroup}, W::WeylGroup)`](@ref)
+can be used instead.
+"""
+function FPGroup(W::WeylGroup)
+  return codomain(isomorphism(FPGroup, W))
+end
+
+@doc raw"""
+    PermGroup(W::WeylGroup) -> PermGroup
+    permutation_group(W::WeylGroup) -> PermGroup
+
+Construct a group of type `PermGroup` that is isomorphic to `W`.
 
-Also see: [`isomorphism(::Type{FPGroup}, ::WeylGroup)`](@ref).
+If one needs the isomorphism then [`isomorphism(::Type{PermGroup}, W::WeylGroup)`](@ref)
+can be used instead.
 """
-function fp_group(W::WeylGroup; set_properties::Bool=true)
-  return codomain(isomorphism(FPGroup, W; set_properties))
+function PermGroup(W::WeylGroup)
+  return codomain(isomorphism(PermGroup, W))
 end
 
+fp_group(W::WeylGroup) = FPGroup(W)
+permutation_group(W::WeylGroup) = PermGroup(W)
+
 @doc raw"""
     isomorphism(::Type{FPGroup}, W::WeylGroup) -> Map{WeylGroup, FPGroup}
 
-Construct an isomorphism between `W` and a group of type `FPGroup`.
+Return an isomorphism from `W` to a group `H` of type `FPGroup`.
 
-The properties of the codomain group and the isomorphism are described in [`fp_group(::WeylGroup)`](@ref).
+`H` will be the quotient of a free group with the same rank as `W`,
+where we have the natural 1-to-1 correspondence of generators, modulo the Coxeter relations of `W`.
+
+Isomorphisms are cached in `W`, subsequent calls of `isomorphism(FPGroup, W)` yield identical results.
+
+If only the image of such an isomorphism is needed, use `fp_group(W)`.
 """
-function isomorphism(::Type{FPGroup}, W::WeylGroup; set_properties::Bool=true)
+function isomorphism(T::Type{FPGroup}, W::WeylGroup; on_gens::Bool=true)
+  on_gens = true # we ignore the on_gens flag, the iso will *always* map gens onto gens
+  isos =
+    get_attribute!(Dict{Tuple{Type,Bool},Any}, W, :isomorphisms)::Dict{Tuple{Type,Bool},Any}
+  return get!(isos, (T, on_gens)) do
+    G = _isomorphic_group_on_gens(T, W)
+
+    # help GAP a bit
+    set_is_finite(G, is_finite(W))
+    is_finite(W) && set_order(G, order(W))
+
+    iso = function (w::WeylGroupElem)
+      return G([i => 1 for i in word(w)])
```

Comment:
```suggestion
      return G(syllables(w))
```

Reply:
```suggestion
      return G(syllables(w)) # TODO: change to letters once G supports that input
```
Good catch. Let me add a reminder to adapt this to use letters once @jamesnohilly adds such a constructor for FPGroups